pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -187,6 +186,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			r.Recorder.Event(app, event.Warning(velatypes.ReasonFailedWorkflow, err))
 			return r.endWithNegativeCondition(logCtx, app, condition.ErrorCondition(common.WorkflowCondition.String(), err), common.ApplicationRunningWorkflow)
 		}
+		app.Status.SetConditions(condition.ReadyCondition(common.RenderCondition.String()))
```

Comment:
Is it consistent with Line187? If GenerateApplicationSteps failed, it would not mark the RenderCondition failed right?

Reply:
Yes, if workflow generates steps failed, the status should be rendering